kind/pkg/cluster/config/encoding/encoding_test.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package encoding
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/test-infra/kind/pkg/cluster/config"
+)
+
+// TODO(bentheelder): once we have multiple config API versions we
+// will need more tests for Load and LoadCurrent
+
+func TestLoadCurrent(t *testing.T) {
+	cases := []struct {
+		Name        string
+		Path        string
+		ExpectError bool
+	}{
+		{
+			Name:        "valid minimal",
+			Path:        "./testdata/valid-minimal.yaml",
+			ExpectError: false,
+		},
+		{
+			Name:        "invalid path",
+			Path:        "./testdata/not-a-file.bogus",
+			ExpectError: true,
+		},
+		{
+			Name:        "invalid apiVersion",
+			Path:        "./testdata/invalid-apiversion.yaml",
+			ExpectError: true,
+		},
+		{
+			Name:        "invalid yaml",
+			Path:        "./testdata/invalid-yaml.yaml",
+			ExpectError: true,
+		},
+	}
```

Comment:
Missing coverage for lifecycle hooks fields

Reply:
yeah, also missing implementation for these outside of the config. I'll fix this

Reply:
now has coverage for lifecycle hooks